src/ios/PurchasesPlugin.swift

```diff
@@ -1,10 +1,397 @@
+
+@objc public class CDVPurchasesPlugin : CDVPlugin {
+
+    private var updatedPurchaserInfoCallbackID: String?
+    private var shouldPurchasePromoProductCallbackID: String?
+    private var defermentBlocks: [RCDeferredPromotionalPurchaseBlock] = []
+
+    @objc(setupPurchases:) func setupPurchases(command: CDVInvokedUrlCommand) {
+        let apiKey = command.arguments[0] as? String ?? ""
+        let appUserID = command.arguments[1] as? String
+        let observerMode = command.arguments[2] as? NSNumber ?? NSNumber(value: false)
+        let userDefaultsSuiteName = command.arguments[3] as? String
+
+        Purchases.configure(withAPIKey: apiKey,
+                            appUserID: appUserID,
+                            observerMode: observerMode.boolValue,
+                            userDefaultsSuiteName: userDefaultsSuiteName,
+                            platformFlavor: self.platformFlavor,
+                            platformFlavorVersion: self.platformFlavorVersion)
+        Purchases.shared.delegate = self
+
+        self.updatedPurchaserInfoCallbackID = command.callbackId
+        let pluginResult = CDVPluginResult(status: .noResult)
+        pluginResult?.setKeepCallbackAs(true)
+        self.commandDelegate.send(pluginResult, callbackId: command.callbackId)
+    }
+
+    @objc(setAllowSharingStoreAccount:) func setAllowSharingStoreAccount(command: CDVInvokedUrlCommand) {
+        let allowSharingStoreAccount = command.arguments[0] as? NSNumber ?? NSNumber(value: false)
+
+        RCCommonFunctionality.setAllowSharingStoreAccount(allowSharingStoreAccount.boolValue)
+        self.sendOKFor(command: command)
+    }
+
+    @objc(addAttributionData:) func addAttributionData(command: CDVInvokedUrlCommand) {
+        let network = command.arguments[1] as? NSNumber
+        let networkUserId = command.arguments[2] as? NSString
+
+        guard let data = command.arguments[0] as? NSDictionary? as? [AnyHashable: Any],
+              let network = network?.intValue,
+              let networkUserId = networkUserId as? String else {
+            self.sendBadParametersFor(command: command,
+                                      parametersNamed: ["data", "network", "networkUserId"],
+                                      expectedTypes: [NSDictionary.self, NSNumber.self, NSString.self])
+            return
+        }
+
+        RCCommonFunctionality.addAttributionData(data, network: network, networkUserId: networkUserId)
+        self.sendOKFor(command: command)
+    }
+
+    @objc(getOfferings:) func getOfferings(command: CDVInvokedUrlCommand) {
+        RCCommonFunctionality.getOfferingsWithCompletionBlock { offerings, error in
+            // TODO: change to pass `responseCompletion` directly instead of extracting block, this is for debugging.
+            self.responseCompletion(forCommand: command)(offerings, error)
+        }
+    }
+
+    @objc(getProductInfo:) func getProductInfo(command: CDVInvokedUrlCommand) {
```

Comment:
I was making the assumption we needed `@objc` because the underlying framework is objc and I'm assuming it uses some type of reflection to access the methods. I'm going to test it out without this, though.

Reply:
Tested, and yeah we need `@obj`

Reply:
makes sense, thanks for looking into it!